include/nix/file/DirectoryWithAttributes.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2013 - 2015, German Neuroinformatics Node (G-Node)
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted under the terms of the BSD License. See
+// LICENSE file in the root of the Project.
+
+#ifndef NIX_DIRECTORYWITHATTRIBUTES_HPP
+#define NIX_DIRECTORYWITHATTRIBUTES_HPP
+
+#include <boost/filesystem.hpp>
+#include <nix/file/AttributesFS.hpp>
+#include <nix/file/Directory.hpp>
+
+#include <string>
+#include <vector>
+
+namespace nix {
+namespace file {
+
+class DirectoryWithAttributes: public Directory {
+
+private:
+    mutable AttributesFS attributes;
```

Comment:
Do we have a const-correctness issue somewhere, because `mutable` should really be avoided.